lib/account.js

```diff
@@ -2,269 +2,295 @@ var Boom = require('boom');
+      headers: {
+        'Content-Type': 'application/json',
+        'x-ratelimit-ip': getIPAddress(request)
+      },
+      uri: loginAPI + '/api/v2/user/create'
+    });
 
-      resetRequest.on('response', function(message) {
-        parseMessage(message, function(err, json) {
-          callback(err, json);
-        });
-      });
+    createRequest.on('error', callback);
 
-      resetRequest.end(JSON.stringify({
-        uid: request.payload.uid,
-        resetCode: request.payload.resetCode,
-        newPassword: request.payload.password
-      }), 'utf8');
-    },
-    createUser: function(request, callback) {
-      var createRequest = hyperquest({
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-          'x-ratelimit-ip': getIPAddress(request)
-        },
-        uri: loginAPI + '/api/v2/user/create'
+    createRequest.on('response', function(message) {
+      parseMessage(message, function(err, json) {
+        callback(err, json);
       });
+    });
 
-      createRequest.on('error', callback);
+    createRequest.end(JSON.stringify({
+      user: {
+        email: request.payload.email,
+        username: request.payload.username,
+        mailingList: request.payload.feedback,
+        prefLocale: request.payload.lang
+      },
+      oauth: {
+        client_id: request.payload.client_id
+      },
+      password: request.payload.password,
+      audience: options.uri
+    }), 'utf8');
+  });
+
+  server.method('account.getUser', function(userId, callback) {
+    var ttl = null;
+    var getRequest = hyperquest({
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      uri: loginAPI + '/user/id/' + userId
+    });
 
-      createRequest.on('response', function(message) {
-        parseMessage(message, function(err, json) {
-          callback(err, json);
-        });
-      });
+    getRequest.on('error', function(err) {
+      callback(err, null, 0);
+    });
+
+    getRequest.on('response', function(message) {
+      parseMessage(message, function(err, json) {
+        if (json && !json.user) {
+          ttl = 0;
+        }
 
-      createRequest.end(JSON.stringify({
-        user: {
-          email: request.payload.email,
-          username: request.payload.username,
-          mailingList: request.payload.feedback,
-          prefLocale: request.payload.lang
-        },
-        oauth: {
-          client_id: request.payload.client_id
-        },
-        password: request.payload.password,
-        audience: config.uri
-      }), 'utf8');
-    },
-    getUser: function(userId, callback) {
-      var getRequest = hyperquest({
-        method: 'GET',
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        uri: loginAPI + '/user/id/' + userId
+        callback(err, json, ttl);
       });
+    });
+  }, {
+    cache: {
+      segment: 'accounts.getUser',
+      expiresIn: 1000 * 60 * 60 * 24,
```

Comment:
Lets set this to something shorter like 60 seconds (so 60 * 1000), and staleIn to 30 seconds. That should be long enough for a burst series of requests.